node/tests/GlideClient.test.ts

```diff
@@ -979,6 +979,76 @@ describe("GlideClient", () => {
+        async (protocol) => {
+            // Create a client configuration with a generous request timeout
+            const config = getClientConfigurationOption(
+                cluster.getAddresses(),
+                protocol,
+                { requestTimeout: 20000 }, // Long timeout to allow debugging operations (sleep for 7 seconds)
+            );
+
+            // Initialize the primary client
+            const client = await GlideClient.createClient(config);
+
+            try {
+                // Run a long-running DEBUG SLEEP command using the first client (client)
+                const debugCommandPromise = client.customCommand(
+                    ["DEBUG", "sleep", "7"], // Sleep for 7 seconds
+                );
+
+                // Function that tries to create a client with a short connection timeout (100ms)
+                const failToCreateClient = async () => {
+                    await new Promise((resolve) => setTimeout(resolve, 1000)); // Wait for 1 second before retry
+                    await expect(
+                        GlideClient.createClient({
+                            connectionBackoff: {
+                                exponentBase: 2,
+                                factor: 100,
+                                numberOfRetries: 1,
+                            },
+                            advancedConfiguration: { connectionTimeout: 100 }, // 100ms connection timeout
+                            ...config, // Include the rest of the config
+                        }),
+                    ).rejects.toThrowError(/timed out/i); // Ensure it throws a timeout error
+                };
+
+                // Function that verifies that a larger connection timeout allows connection
+                const connectWithLargeTimeout = async () => {
+                    await new Promise((resolve) => setTimeout(resolve, 1000)); // Wait for 1 second before retry
+                    const longerTimeoutClient = await GlideClient.createClient({
+                        connectionBackoff: {
+                            exponentBase: 2,
+                            factor: 100,
+                            numberOfRetries: 1,
+                        },
+                        advancedConfiguration: { connectionTimeout: 10000 }, // 10s connection timeout
+                        ...config, // Include the rest of the config
+                    });
+                    expect(await client.set("x", "y")).toEqual("OK");
+                    longerTimeoutClient.close(); // Close the client after successful connection
+                };
+
+                // Run both the long-running DEBUG SLEEP command and the client creation attempt in parallel
+                await Promise.all([
+                    debugCommandPromise, // Run the long-running command
+                    failToCreateClient(), // Attempt to create the client with a short timeout
```

Comment:
You can run all 3 promises as you do in java tests

Reply:
wasnt sure how it would behave as single thread lang